core-plugins/src/main/java/io/cdap/plugin/batch/action/EmailAction.java

```diff
@@ -107,37 +109,39 @@ public Config() {
       includeWorkflowToken = false;
     }
 
-    public void validate() {
-      super.validate();
+    public void validate(FailureCollector collector) {
+      super.validate(collector);
 
       if (!containsMacro("username") && (Strings.isNullOrEmpty(username) ^ Strings.isNullOrEmpty(password))) {
-        throw new IllegalArgumentException("You must either set both username and password or " +
-                                             "neither username nor password.");
+        collector.addFailure("You must either set both username and password or neither username nor password.",
+                  "Leave username and password fields empty or provide values for both fields")
+                  .withConfigProperty("username").withConfigProperty("password");
       }
 
       if (!containsMacro("sender")) {
         try {
           InternetAddress[] addresses = InternetAddress.parse(sender);
           if (addresses.length == 0) {
-            throw new IllegalArgumentException("Must specify a sender email address.");
+            collector.addFailure("Sender email was not specified.", "Provide a sender email address.")
```

Comment:
When we complain that some required property is missing, we can just use a single error message with no corrective action.

"Sender email must be specified."

should be good enough

Reply:
Removed corrective action for all missing property errors